### ISSUE TITLE
Accessing parts of vectors

### ISSUE THREAD

**Issue by xix-xeaon** [godot-nim/gdext-nim#109]
It confused me a lot trying to do `v.x` etc until I found the swizzling operator `.*` and realized that that's the only way of accessing the parts (besides `v[0]` of course). I'd like to add the following for x, y, z and w to `geometrics/typedef.nim` (I can create a PR):

```nim
template x[N: static int, T](v: array[N, T]): T = v[0]
template `x=`[N, T](v: var array[N, T], val: T) = v[0] = val
```

**Comment by panno8M** [comment]
Certainly this is needed, hope you create a PR.

**Comment by panno8M** [comment]
Speaking of which, since you have write access, you can clone gdext-nim directly and push the branch.

You can also create a PR from a fork, but it is easier to create one from a checked out topic branch.

**Comment by xix-xeaon** [comment]
There was some kind of authorization issue when I tried to publish a new branch for this repository from GitHub Desktop, so I forked and branched instead.

**Comment by xix-xeaon** [comment]
Implemented in #116.
